homeassistant/components/openuv/coordinator.py

```diff
@@ -114,10 +114,11 @@ async def _async_update_data(self) -> dict[str, Any]:
         """Fetch data from OpenUV."""
         try:
             data = await self.update_method()
-        except InvalidApiKeyError:
+        except InvalidApiKeyError as err:
             await self._invalid_api_key_monitor.async_increment()
+            raise UpdateFailed(f"Error during data update: {err}") from err
```

Comment:
`UpdateFailed` already indicates that updating the data failed, so you could just stringify err if it is a human readable error.
```suggestion
            raise UpdateFailed(str(err)) from err
```